--- bash/lib/readline/display.c	2023-11-07 19:54:02
+++ lib/readline/display.c	2023-11-07 18:58:36
@@ -3581,3 +3581,12 @@
   return width;
 }
 #endif /* HANDLE_MULTIBYTE */
+
+/* For Bash Commander: return the current cursor column number. */
+int
+_rl_cursor_col ()
+{
+  if ((MB_CUR_MAX == 1 || rl_byte_oriented) && _rl_last_v_pos == 0 && visible_wrap_offset > 0)
+    return _rl_last_c_pos - visible_wrap_offset;
+  return _rl_last_c_pos;
+}
